src/fixture/fixture.ts

```diff
@@ -3,14 +3,26 @@ import type { Context } from '../transformer/generator';
 import { Transformer } from '../transformer/transformer';
 import { fixtureGenerators } from './generators';
 
-export class Fixture extends Transformer {
-	generators = fixtureGenerators;
-
+export interface Fixture extends Transformer {
```

Comment:
It's just extending the interface of `class Fixture extends Transformer {}` to tell Typescript that it's `z.infer` instead of `unknown`. I thought it was a cleaner way of defining the type that's output. It doesn't have anything to do with the missing error update.

```ts
export interface Fixture extends Transformer {
	// explicitly define the return type
	from<TSchema extends z.ZodTypeAny>(
		schema: TSchema,
		context?: Context
	): z.infer<TSchema>;
}

export class Fixture extends Transformer {
	generators = fixtureGenerators;
}
```

Reply:
My bad, I meant what does this have to be an interface in comparison to just having the Fixture class?

Reply:
That's what I originally had, but then I realized we're just wrapping a function purely for the sake of typing it correctly. This is just adding code and complexity to the runtime that's not needed.

```ts
export class Fixture extends Transformer {
	generators = fixtureGenerators;
	
	from<TSchema extends z.ZodTypeAny>(
		schema: TSchema,
		context?: Context
	): z.infer<TSchema> (schema, context) {
		return super.from(schema, context); // this feels totally unnecessary
	}
}
```

I DO understand, however, if you'd like to keep this because it's easier for people to understand. Totally your call here.

Or maybe just use the `super` for documentation, but keep the pure typing internally for Fixture to reduce code size and runtime complexity?

Reply:
I find myself pushing to keep it this way. I really hate to introduce runtime cruft to modify static typing.

I'd be willing to use `super.from` in the documentation, though...

I wish there was a better way to do this 😫